src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/PrivateDataRetrievalService.java

```diff
@@ -53,8 +56,13 @@ private ManagingOfficerListDataApi getManagingOfficerData(String companyNumber)
 
       ApiLogger.debug("Retrieving Managing Officer data for Company Number ", logMap);
       return managingOfficerDataList;
-
-    } catch (URIValidationException | IOException e) {
+    } catch (ApiErrorResponseException e) {
+      if (e.getStatusCode() == HttpServletResponse.SC_NOT_FOUND) {
+        ApiLogger.errorContext("No Managing Officers data found for Company Number "+companyNumber, e);
```

Comment:
please use  ApiLogger.error(message, e, logMap); like line 67 so that we can search logs for company number key.